src/models/mSeriesChallenge.ts

```diff
@@ -0,0 +1,268 @@
+  }
+}
+
+export class MSeriesChallenge extends Model {
+  private _mSeriesChallengeData: MSeriesChallengeData
+  private _mSeriesChallengeParticipant?: MSeriesChallengeParticipant
+
+  constructor (mSeriesChallengeData: MSeriesChallengeData, sessionHandler: SessionHandler) {
+    super(sessionHandler)
+    this._mSeriesChallengeData = mSeriesChallengeData
+  }
+
+  private setMSeriesChallenge (mSeriesChallengeData: MSeriesChallengeData) {
+    this._mSeriesChallengeData = mSeriesChallengeData
+  }
+
+  private setMSeriesChallengeParticipant (mSeriesChallengeParticipant: MSeriesChallengeParticipant) {
+    this._mSeriesChallengeParticipant = mSeriesChallengeParticipant
+  }
+
+  async reload (options: { includeOwnedParticipantData: boolean } = { includeOwnedParticipantData: false }) {
+    const { mSeriesChallenge } = await this.action('mSeriesChallenge:show', { id: this._mSeriesChallengeData.id, userId: this.sessionHandler.userId }) as MSeriesChallengeResponse
+
+    if (options.includeOwnedParticipantData) {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:show', { mSeriesChallengeId: this._mSeriesChallengeData.id, userId: this.sessionHandler.userId }) as MSeriesChallengeParticipantResponse
+      this.setMSeriesChallengeParticipant(new MSeriesChallengeParticipant(mSeriesChallengeParticipant, this.sessionHandler))
+    }
+
+    this.setMSeriesChallenge(mSeriesChallenge)
+
+    return this
+  }
+
+  /**
+   * Method to end a challenge. Only challenge owners can end a challenge.
+   */
+  async end () {
+    await this.action('mSeriesChallenge:delete', { id: this._mSeriesChallengeData.id })
+  }
+
+  /**
+   * Method to leave a challenge. Only challenge participants can leave a challenge.
+   */
+  async leave () {
+    await this.action('mSeriesChallengeParticipant:delete', { mSeriesChallengeId: this._mSeriesChallengeData.id, userId: this.sessionHandler.userId })
+  }
+
+  /**
+   * Method to join a challenge. Challenge instance must contain joinCode to successfully join.
+   */
+  async join () {
+    if (this._mSeriesChallengeData.joinCode !== undefined) {
+      const { mSeriesChallengeParticipant } = await this.action('mSeriesChallengeParticipant:create', { joinCode: this._mSeriesChallengeData.joinCode, userId: this.sessionHandler.userId }) as MSeriesChallengeParticipantResponse
+      const p = new MSeriesChallengeParticipant(mSeriesChallengeParticipant, this.sessionHandler)
+      this.setMSeriesChallengeParticipant(p)
+      return p
+    }
+    return undefined
+  }
+
+  /**
+   * This method can return the current session user's participant data or a different user's participant data.
+   *
+   * @param mSeriesChallengeParticipantId Id of participant to be shown. Leave undefined for current session user's participant data.
+   * @returns An MSeries Challenge Participant
+   */
+  async getParticipant (mSeriesChallengeParticipantId?: number) {
```

Comment:
This split functionality is also weird and will cause issues in the long-run. Unify the functionality so you aren't querying the same data 2 different ways.

Reply:
Are you saying to do something like this:

```
async getOwnedParticipantData () { ... }
async getParticipant (mSeriesChallengeParticipantId: number) { ... }
```

or are you saying to change to `mSeriesChallengeParticipant:show` action params. Right now the action accepts either:
- `mSeriesChallengeId` for getting the current session user's participant data
- OR
- `mSeriesChallengeParticipantId` for getting a different user's participant data

So there are inherently two different ways to get the "same" data.